var/spack/repos/builtin/packages/rose/package.py

```diff
@@ -85,17 +110,82 @@ def languages(self):
 
     def configure_args(self):
         spec = self.spec
-        cc = self.compiler.cc
-        cxx = self.compiler.cxx
-        return [
+
+        if '+mvapich2_backend' in spec:
+            cc = spec['mvapich2'].mpicc
+            cxx = spec['mvapich2'].mpicxx
+        else:
+            cc = spack_cc
+            cxx = spack_cxx
+
+        if spec.satisfies('@0.9.8:'):
+            edg = "4.12"
+        else:
+            edg = "4.9"
+
+        args = [
             '--disable-boost-version-check',
+            '--enable-edg_version={0}'.format(edg),
             "--with-alternate_backend_C_compiler={0}".format(cc),
             "--with-alternate_backend_Cxx_compiler={0}".format(cxx),
             "--with-boost={0}".format(spec['boost'].prefix),
             "--enable-languages={0}".format(",".join(self.languages)),
-            "--with-z3={0}".format(spec['z3'].prefix) if '+z3' in spec else '',
-            '--disable-tests-directory' if '+tests' not in spec else '',
-            '--enable-tutorial-directory={0}'.format('no'),
         ]
 
-    install_targets = ["install-core"]
+        if '+z3' in spec:
+            args.append("--with-z3={0}".format(spec['z3'].prefix))
+        else:
+            args.append("--without-z3")
+
+        if '+tests' in spec:
+            args.append("--enable-tests-directory")
+        else:
+            args.append("--disable-tests-directory")
+
+        if '+tutorial' in spec:
+            args.append("--enable-tutorial-directory")
+        else:
+            args.append("--disable-tutorial-directory")
+
+        if '+java' in spec:
+            args.append("--with-java={0}".format(spec['java'].prefix))
+        else:
+            args.append("--without-java")
+
+        if '+debug' in spec:
+            args.append("--with-CXX_DEBUG=-g")
+        else:
+            args.append("--without-CXX_DEBUG")
+
+        if '+optimized' in spec:
+            args.append("--with-C_OPTIMIZE=-O0")
+            args.append("--with-CXX_OPTIMIZE=-O0")
+        else:
+            args.append("--without-C_OPTIMIZE")
+            args.append("--without-CXX_OPTIMIZE")
+
+        return args
+
+    def build(self, spec, prefix):
+        # Spack will automatically pass ncpus as the number of make jobs.
+        #
+        # If you really want to srun this on a separate node, you can do this:
+        #
+        #   $ srun -n1 spack install -j16 rose
```

Comment:
The j16 isn't actually necessary, and likely if you `srun` this you want Spack to just use ncpus, which it does automatically, so `srun -n1 spack install rose` is probably sufficient.  I'm assuming ROSE doesn't support `srun` with `-n` > 1?

Reply:
Updated. Thanks Todd!

Reply:
Just happened to notice this after the reference, but you mean `-N` or `-N --exclusive` right?  I assume you want a single process on a full node rather than a single process only reserving one core?

Reply:
Yes actually that's correct.  I'm forgetting that you actually only get 1 core with `-n1`.

Reply:
Great, thanks @trws @tgamblin